engine/src/main/java/org/terasology/rendering/dag/WireframeTask.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2016 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.rendering.dag;
+
+import org.lwjgl.opengl.GL11;
+import static org.lwjgl.opengl.GL11.GL_FILL;
+import static org.lwjgl.opengl.GL11.GL_FRONT_AND_BACK;
+import static org.lwjgl.opengl.GL11.GL_LINE;
+
+/**
+ *
+ */
+public class WireframeTask implements PipelineTask { // TODO: separate this into SetWireframeOnTask and SetWireframeOffTask?
```

Comment:
Nononono. This is a bigger can of worms than you think. =)

As you rightly point that IF block in process() should be avoided.  The way to do this is to make the RenderingConfig and the DebugRenderingConfig _observable_. So, interested Nodes _subscribe_ to changes in those configs and get notified if the changes occur. 

Initially it could be as simple as all subscribers get notified that _something_ has changed and they are responsible for finding out what it is and take action. Eventually subscribers might get notified if only a particular setting is modified.

Didn't I mention  that wireframe was a bit of a funny state change to deal with for starters? Could I recommend to keep the isWireframe() check in the process() method for the time being, working with simpler state changes to start with? You could perhaps implement the state change to handle FBO/Display binding. That looks relatively straightforward to me and it would delay having to make the configs observable.